File: Java/0454-四数相加 II.java
class Solution {
  public int fourSumCount(int[] nums1, int[] nums2, int[] nums3, int[] nums4) {
    Map<Integer, Integer> map = new HashMap<>();
    final int n = nums1.length;
    for (int i = 0; i < n; i++) {
      final int num = nums1[i];
      for (int k = 0; k < n; k++) {
        final int sum = num + nums2[k];
        map.put(sum, map.getOrDefault(sum, 0) + 1);
      }
    }
    int result = 0;
    for (int i = 0; i < n; i++) {
      final int num = nums3[i];
      for (int k = 0; k < n; k++) {
        result += map.getOrDefault(-(num + nums4[k]), 0);
      }
    }
    return result;
  }
}

File: TypeScript/0454-四数相加 II.ts
function fourSumCount(nums1: number[], nums2: number[], nums3: number[], nums4: number[]): number {
  const twoSumMap: Map<number, number> = new Map<number, number>(),
    n: number = nums1.length;
  for (let i = 0; i < n; i++) {
    const num: number = nums1[i];
    for (let k = 0; k < n; k++) {
      const sum = num + nums2[k];
      twoSumMap.set(sum, (twoSumMap.get(sum) || 0) + 1);
    }
  }
  let result: number = 0;
  for (let i = 0; i < n; i++) {
    const num: number = nums3[i];
    for (let k = 0; k < n; k++) {
      const sum = num + nums4[k];
      result += twoSumMap.get(0 - sum) || 0;
    }
  }
  return result;
}
